tactics/hints.mli

```diff
@@ -162,6 +162,7 @@ module Hint_db :
     val cut : t -> hints_path
 
     val unfolds : t -> Id.Set.t * Cset.t
+    val compare: t -> t -> int
```

Comment:
This should come with a comment. And since this function was not exposed before, wouldn't it be preferable to use a ADT as return type?

Reply:
IMO the standard library of OCaml is not a good source of inspiration for how things should be done. I was thinking of a return type of the form `Lt | Eq | Gt`. Of course, I'm also OK with keeping things as they are, as long as the comment explains how hints are compared by this function.